src/main/java/com/backend/controller/PortfolioController.java

```diff
@@ -189,6 +191,33 @@ public GetAllAssetsByPortfolioIdResponse getAllAssetsByPortfolioId(@PathVariable
 		return response;
 	}
 
+	@GetMapping(path = "/portfolio/{pid}/transactions")
+	public List<Map<String, Object>> getTransactionsByPortfolioId(@PathVariable int pid) {
+		List<PortfolioAsset> portfolioAssetList = portfolioAssetService.findAllByPortfolioId(pid);
+		List<Map<String, Object>> transactionList = new ArrayList<>();
+
+		for (int i = 0; i < portfolioAssetList.size(); i++) {
```

Comment:
suggestion: 

can do
```java
for (PortfolioAsset asset: portfolioAssetList) {
    // ... snipped ...
    asset.getDateCreated();  // access `asset`
    // ... snipped ...
}
```

instead of 
```java
for (int i = 0; i < portfolioAssetList.size(); i++t) {
    // ... snipped ...
    portfolioAssetList.get(i).getDateCreated();  // access `asset`
    // ... snipped ...
}
```

since the index `i` is not being used aside from accessing elements and the for-each loop would be less verbose and easier to follow